### PR TITLE
Simplify token vector's management in SourceFile, NFC

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test and merge

File: include/swift/AST/Module.h [hunk 1]
```diff
@@ -1073,23 +1073,22 @@ class SourceFile final : public FileUnit {
...
-    return EnabledAndAllCorrectedTokens.first;
+    return (bool)pAllCorrectedTokens;
   }
 
 private:
-  /// Whether the SourceFile instance opts to collect underlying tokens and
-  /// the vector containing these tokens if so.
-  std::pair<bool, std::vector<Token>> EnabledAndAllCorrectedTokens;
+  /// If not null, the pointee vector should contain tokens of this source file.
+  std::unique_ptr<std::vector<Token>> pAllCorrectedTokens;
```
Comment (jrose-apple) [inline include/swift/AST/Module.h]: Do you need the extra level of indirection, or can you just use Optional here?

Comment (nkcsgexi) [reply include/swift/AST/Module.h]: Yes, we do need the extra level. Is Optional any better than the unique pointer?

Comment (jrose-apple) [reply include/swift/AST/Module.h]: That's what I meant by indirection, sorry. `std::unique_ptr` creates a separate allocation, while `Optional` reserves storage inline. I guess I don't actually know how big a `std::vector` is but I don't think it's going to make SourceFile unnecessarily large, and less malloc traffic is usually a good thing.

**Pull request merged**